File: hacking/src/renderer/apiService/mappers/subnetwork.ts
import { SubnetworkType } from '../types';

export default function mapUser(data: any): SubnetworkType {
  return {
    id: data.id,
    name: data.name,
    users: data.users,
    networkId: data.networkId,
    firewall: data.firewall,
    operatingSystem: data.operatingSystem,
    ice: data.ice || [],
    accessPoint: data.accessPoint,
    pastHacks: data.pastHacks || [],
  };
}

File: backendDotnet/DatabaseSerializer/SerializededModels/Networks/Subnetwork.cs
namespace Giger.SerializededModels.Networks
{
    public class Subnetwork : Models.Networks.Subnetwork
    {
        public Subnetwork()
        {
            Id = "SN999";
            Name = "Test Subnetwork";
            NetworkId = "N666";
            Users = ["123456", "123457"];
            Firewall = Models.Networks.Firewall.EncryptGuard;
            OperatingSystem = Models.Networks.OperatingSystem.ForceField;
            Ice = [Models.Networks.Ice.ToBeProvided];
            PastHacks = ["attempted hack1", "123457" ];
        }
    }
}
